src/rgw/rgw_json_enc.cc

```diff
@@ -1494,3 +1494,80 @@ void RGWOrphanSearchState::dump(Formatter *f) const
   encode_json("stage", stage, f);
   f->close_section();
 }
+
+void RGWObjTags::dump(Formatter *f) const
+{
+  for (const auto& tag: tag_map){
+    f->open_object_section("tag_map");
+    f->dump_string("Key", tag.first);
+    f->dump_string("Value", tag.second);
+    f->close_section();
+  }
+}
+
+void lc_op::dump(Formatter *f) const
+{
+  encode_json("status", status, f);
+  encode_json("dm_expiration", dm_expiration, f);
+  encode_json("expiration", expiration, f);
+  encode_json("noncur_expiration", noncur_expiration, f);
+  encode_json("mp_expiration", mp_expiration, f);
+  if (expiration_date != boost::none) {
+    utime_t ut(*expiration_date);
+    encode_json("expiration_date", ut, f);
+  }
+  if (obj_tags != boost::none) {
+    encode_json("obj_tags", *obj_tags, f);
+  }
+}
+
+void LCFilter::dump(Formatter *f) const
+{
+  encode_json("prefix", prefix, f);
+  encode_json("obj_tags", obj_tags, f);
+}
+
+void LCExpiration::dump(Formatter *f) const
+{
+  encode_json("days", days, f);
+  encode_json("date", date, f);
+}
+
+void LCRule::dump(Formatter *f) const
+{
+  encode_json("id", id, f);
+  encode_json("prefix", prefix, f);
+  encode_json("status", status, f);
+  encode_json("expiration", expiration, f);
+  encode_json("noncur_expiration", noncur_expiration, f);
+  encode_json("mp_expiration", mp_expiration, f);
+  encode_json("filter", filter, f);
+  encode_json("dm_expiration", dm_expiration, f);
+}
+
+void RGWLifecycleConfiguration::dump(Formatter *f) const
+{
+  map<string, lc_op>::const_iterator iter = prefix_map.begin();
+  f->open_array_section("prefix_map");
+  for (; iter != prefix_map.end(); ++iter) {
+    f->open_object_section("entry");
+    f->dump_string("prefix", iter->first);
+    f->open_object_section("lc_op");
+    iter->second.dump(f);
+    f->close_section();
+    f->close_section();
+  }
+  f->close_section();
+
+  multimap<string, LCRule>::const_iterator multi_iter = rule_map.begin();
+  f->open_array_section("rule_map");
+  for (; multi_iter != rule_map.end(); ++multi_iter) {
+    f->open_object_section("entry");
+    f->dump_string("id", multi_iter->first);
+    f->open_object_section("rule");
+    multi_iter->second.dump(f);
+    f->close_section();
+    f->close_section();
+  }
```

Comment:
it's a matter of taste, but you can also do
<pre>
+  f->open_object_section("rule_map");
 +  for (; multi_iter != rule_map.end(); ++multi_iter) {
 +    f->open_object_section(multi_iter->first.c_str());
 +    multi_iter->second.dump(f);
 +    f->close_section();
 +  }
 +  f->close_section();
</pre>

Reply:
or even
<pre>
+  f->open_object_section("rule_map");
 +  for (; multi_iter != rule_map.end(); ++multi_iter) {
 +    f->dump_object(multi_iter->first.c_str(), multi_iter->second);
 +  }
 +  f->close_section();
</pre>

Reply:
@liewegas  Thank you for your suggestion, I have fixed it. Please check,  when you get a chance.